src/Database/Concerns/ResourceSyncing.php

```diff
@@ -43,4 +46,15 @@ public function shouldSync(): bool
     {
         return true;
     }
+
+    public function tenants(): MorphToMany
+    {
+        return $this->morphToMany($this->getResourceTenantModelName(), 'tenant_resources', 'tenant_resources', 'resource_global_id', 'tenant_id', 'global_id')
+            ->using(TenantMorphPivot::class);
+    }
```

Comment:
Why does this forcibly use `MorphToMany`?

Reply:
Using the `polymorphic` approach is the default in this PR (We discussed this). So with this approach, I think it's okay to use `MorphToMany` here. And developers can override this method of course like 
```php
public function tenants(): BelongsToMany
{
    return $this->belongsToMany(ResourceTenant::class, 'tenant_users', 'global_user_id', 'tenant_id', 'global_id')
        ->using(TenantPivot::class);
}
```

Is there something wrong with this return type?

Reply:
I think we had a discussion somewhere where we decided that `MorphToMany` is fine. And it's `BelongsToMany` can be used too because the `parent` return type is the same.

Reply:
> And it's BelongsToMany can be used too because the parent return type is the same.

What do you mean by this?

Reply:
> What do you mean by this?

This method has a `MorphToMany` return type. Its base class is `BelongsToMany`. So in case, someone wants to use the separate table approach (opposite of polymorphic), `BelongsToMany` can be specified in the overridden method.

Reply:
I think the base class doesn't matter here in any way. When you have a case where "A extends B", your typehint says A, and you want to change it to B later on, then PHP won't let you. The child class can have additional things, it's a separate type. Using a "B" typehint would let you use both A and B.

That said, this is a trait so I see your point about MorphToMany being the default, and about people being able to override this.

Do we override the method with a BelongsToMany in any tests? We should cover this, so that we know that overriding the method works fine.

Reply:
> I think the base class doesn't matter here in any way. When you have a case where "A extends B", your typehint says A, and you want to change it to B later on, then PHP won't let you. The child class can have additional things, it's a separate type. Using a "B" typehint would let you use both A and B.

Yes, that makes sense. So trait method type should be `BelongsToManay`, then we can use any type between two. 

> Do we override the method with a BelongsToMany in any tests? We should cover this, so that we know that overriding the method works fine.

Yes, we are overriding this method and providing the `BelongsToMany` return type in `ResourceSyncingTest` tests. 
https://github.com/archtechx/tenancy/blob/144df028a0a21e0c3098d9549604379a0a9571cb/tests/ResourceSyncingTest.php#L852-L856

That's strange then. It should throw an error but it's working.

Reply:
I've updated the trait method return type to `BelongsToMany`, which makes sense. As `BelongsToMany` is the parent type, developer can use both `BelongsToMany` or `MorphToMany`

Reply:
I've updated the trait method return type to `BelongsToMany` too.

Reply:
Why? I said that this is a trait so there should be no issues overriding it. And if:
- the default is `MorphToMany`
- to use `BelongsToMany`, people are expected to override the method

Then it makes no sense to have a `$this->morphTomany()` method that returns `BelongsToMany`.

Reply:
Makes sense.